spec/jasmine-reportportal-reporter.spec.js

```diff
@@ -0,0 +1,309 @@
+            reporter.parentIds = [0, 1];
+
+            const levelType = reporter.getTopLevelType();
+
+            expect(levelType).toBe('TEST');
+        });
+
+        it('should return level type \'suite\' if parentIds is empty', function () {
+            reporter.parentIds = [];
+
+            const levelType = reporter.getTopLevelType();
+
+            expect(levelType).toBe('SUITE');
+        });
+    });
+
+    describe('suiteStarted', function() {
+        it('should send a request to the agent', function() {
+            reporter.additionalCustomParams = { attributes: { key: 'value' } };
+            spyOn(reporter.client, 'startTestItem').and.returnValue({
+                tempId: '3452',
+				promise: Promise.resolve()
+            });
+
+            reporter.suiteStarted({
+                description: 'test description',
+                fullName: 'test name'
+            });
+
+            expect(reporter.client.startTestItem).toHaveBeenCalledWith({
+                type: 'SUITE',
+                description: 'test description',
+                name: 'test name',
+                attributes: { key: 'value' }
+            }, tempLaunchId, null);
+        });
+
+        it('should create an element in parentIds', function() {
+            spyOn(reporter.client, 'startTestItem').and.returnValue({
+                tempId: '3452',
+                promise: Promise.resolve()
+            });
+
+            reporter.suiteStarted({
+                description: 'test description',
+                fullName: 'test name'
+            });
+
+            expect(reporter.parentIds.length).toBe(1);
+        });
+    });
+
+    describe('specStarted', function() {
+        it('should send a request to the agent', function() {
+            spyOn(reporter.client, 'startTestItem').and.returnValue({
+                tempId: '3452',
+                promise: Promise.resolve()
+            });
+
+            reporter.specStarted({
+                description: 'test description',
+                fullName: 'test name'
+            });
+
+            expect(reporter.client.startTestItem).toHaveBeenCalledWith({
+                type: 'STEP',
+                description: 'test description',
+                name: 'test name'
+            }, tempLaunchId, null);
+        });
+
+        it('should call setParentId with the appropriate parameter', function() {
+            spyOn(reporter.client, 'startTestItem').and.returnValue({
+                tempId: '3452',
+                promise: Promise.resolve()
+            });
+            spyOn(reporter, 'setParentId');
+
+            reporter.suiteStarted({
+                description: 'test description',
+                fullName: 'test name'
+            });
+
+            expect(reporter.setParentId).toHaveBeenCalledWith('3452');
+        });
+    });
+
+    describe('specDone', function() {
+        it('should call finishParent, additionalCustomParams should be empty object', function() {
```

Comment:
what about `additionalCustomParams` in this test?

Reply:
inside of the function specDone, this value must be cleaned. So I added the check of it
`expect(reporter.additionalCustomParams).toEqual({});`